internal/controller/storageclaim_controller.go

```diff
@@ -424,6 +476,71 @@ func (r *StorageClaimReconciler) reconcilePhases() (reconcile.Result, error) {
 				if err != nil {
 					return reconcile.Result{}, fmt.Errorf("failed to create or update VolumeSnapshotClass: %s", err)
 				}
+			case "VolumeReplicationClass":
+				data := map[string]string{}
+				err = json.Unmarshal(resource.Data, &data)
+				if err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to unmarshal StorageClaim configuration response: %v", err)
+				}
+				data["replication.storage.openshift.io/replication-secret-namespace"] = r.OperatorNamespace
+				// generate a new clusterID for cephfs subvolumegroup, as
+				// storageclaim is clusterscoped resources using its
+				// hash as the clusterID
+				data["clusterID"] = r.storageClaimHash
+
+				if !r.AvailableCrds[drClusterConfigCRDName] {
+					continue
+				}
+				// sanity check that there's a single DRClusterConfig per cluster
+				drclusterconfigs := &ramenv1alpha1.DRClusterConfigList{}
+				err = r.list(drclusterconfigs)
+				if err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to get DRClusterConfigList: %v", err)
+				}
+				if len(drclusterconfigs.Items) == 0 {
+					continue
+				}
+				if len(drclusterconfigs.Items) > 1 {
+					return reconcile.Result{}, fmt.Errorf("there's more than one DRClusterConfig attached to the cluster")
+				}
+				vrcs := &replicationv1alpha1.VolumeReplicationClassList{}
+				err = r.list(vrcs)
+				if err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to get VolumeReplicationClassList: %v", err)
+				}
+				drclusterconfig := &drclusterconfigs.Items[0]
+				// check if Scheduling intervals have been deleted from DRClusterConfig
+				replicationSchedules := drclusterconfig.Spec.ReplicationSchedules
+				// delete VolumeReplicationClasses that have their respective replication schedule removed from DRClusterConfig
+				for _, vrc := range vrcs.Items {
+					if !slices.Contains(replicationSchedules, vrc.Spec.Parameters["schedulingInterval"]) {
+						err = r.Delete(r.ctx, &vrc)
+						if err != nil {
+							return reconcile.Result{}, fmt.Errorf("failed to delete VolumeReplicationClass %v: %v", vrc.Name, err)
+						}
```

Comment:
I know this will work in go > 1.22, however could you pls don't capture loop var but refer that via index like, as I believe loopvar does a shallow copy.
`for idx := range vrcs.Items { vrc := &vrcs.Items[idx]; .... }`?

could you pls exclude resource not found error for Delete op? like, list is from cache but delete hits the api server and it may be possible in api server the resource is already deleted, just better to code against any race conditions.